yt_dlp/extractor/xumo.py

```diff
@@ -1,38 +1,89 @@
-        'url': 'https://www.xumo.tv/free-movies/zombie-high/XM01OEG1MIJVJR',
-        'md5': '2843529c1abf5f86f61165637f89cbae',
-        'info_dict': {
-            'id': 'XM01OEG1MIJVJR',
+
+    def _generate_video_link(self, video_id):
+        return 'https://www.xumo.tv/free-movies/x/' + video_id
+
+    class ContentType(Enum):
+        SERIES = 'SERIES'
+        EPISODE = 'EPISODIC'
+        MOVIE = 'MOVIE'
+
+    class TestData:
+
+        SERIES_ID = 'XM0AN69OG47PRN'
+        SERIES_TITLE = 'Super Mario World'
+        SERIES_DATA = {
+            'id': SERIES_ID,
+            'title': SERIES_TITLE
+        }
+
+        EPISODE_ID = 'XM02D369HADFRR'
+        EPISODE_MD5 = 'ed2f396272b39f2e0fe47f02b9ae34cc'
+        EPISODE_DATA = {
+            'id': EPISODE_ID,
+            'title': 'Fire Sale // Misadventures In Robin Hood Woods',
+            'ext': 'mp4',
+            'series': SERIES_TITLE,
+            'season_number': 1,
+            'episode_number': 2,
+            'episode': 'Fire Sale // Misadventures In Robin Hood Woods',
+            'description': 'md5:48134d36781cf4b225ec0ee4f05356d3',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'season': 'Season 1',
+            'duration': 1368,
+        }
+
+        MOVIE_ID = 'XM01OEG1MIJVJR'
+        MOVIE_MD5 = '2843529c1abf5f86f61165637f89cbae'
+        MOVIE_DATA = {
+            'id': MOVIE_ID,
             'title': 'Zombie High',
+            'ext': 'mp4',
             'description': 'md5:b620cc0afc18a91fb69275047f6c4d22',
             'duration': 5436,
             'release_year': 1987,
-            'thumbnail': r're:^https?://.*\.jpg$',
-            'ext': 'mp4'
+            'thumbnail': r're:^https?://.*\.jpg$'
         }
+
+    _VALID_URL = r'https?://(?:www\.)?xumo\.tv/.*/(?P<id>XM[A-Z0-9]{12})(?:/.*)?'
+    _TESTS = [{
+        # movie
+        'url': 'https://www.xumo.tv/free-movies/zombie-high/' + TestData.MOVIE_ID,
+        'params': {
+            'check_formats': True
+        },
+        'md5': TestData.MOVIE_MD5,
+        'info_dict': TestData.MOVIE_DATA
     }, {
         # series
-        'url': 'https://www.xumo.tv/tv-shows/super-mario-world/XM0AN69OG47PRN',
-        'playlist_count': 10,
-        'info_dict': {
-            'id': 'XM0AN69OG47PRN',
-            'title': 'Super Mario World'
-        },
+        'url': 'https://www.xumo.tv/tv-shows/super-mario-world/' + TestData.SERIES_ID,
         'params': {
             'skip_download': True
-        }
+        },
+        'playlist_count': 10,
+        'info_dict': TestData.SERIES_DATA
+    }, {
+        # video from network URL
+        'url': 'https://www.xumo.tv/networks/free-movies/99991299/' + TestData.EPISODE_ID,
+        'md5': TestData.EPISODE_MD5,
+        'info_dict': TestData.EPISODE_DATA
+    }, {
+        # episode from generated URL
+        'url': _generate_video_link(NULL, TestData.EPISODE_ID),
+        'md5': TestData.EPISODE_MD5,
+        'info_dict': TestData.EPISODE_DATA
```

Comment:
Revert all of this!

Reply:
What is wrong with it? The previous test cases did not account for individual episodes.

Reply:
Then just fix that. Don't declare all these enums and classes. Look at existing extractors